users/views.py

```diff
@@ -54,6 +54,75 @@ def send_confirm_mail(self, user):
         )
 
 
+def UserRequestResetPasswordView(request):
+    if request.method == "GET":
+        template = loader.get_template('users/user_request_password_reset.html')
+        return HttpResponse(template.render(request=request))
+
+    if request.method == "POST":
+        email = request.POST['email']
+
+        try:
+            user = User.objects.get(email=email)
+            user.password_reset_token = create_hash()
+            user.save()
+
+            # send password reset email with token
+            template = loader.get_template('users/password_reset_email.html')
+            context = Context({'user': user})
+            html = template.render(context)
+
+            send_mail(
+                subject='Password reset',
+                message='hi',
+                from_email='hello@circuit.io',
+                recipient_list=[user.email],
+                fail_silently=False,
+                html_message=html
+            )
+
+            messages.success(request, 'Password reset email send.')
+
+            return redirect('users:login')
+
+        except ObjectDoesNotExist:
+            messages.error(request, 'User with email <{:s}> does not exist'.format(email))
+            return redirect('users:request_password_reset')
+
+
+def UserResetPasswordView(request, token):
+    if request.method == "GET":
+        try:
+            user = User.objects.get(password_reset_token=token)
+            template = loader.get_template('users/user_password_reset.html')
+            return HttpResponse(template.render(request=request))
+        except ObjectDoesNotExist:
+            messages.error(request, "Invalid token.")
```

Comment:
I think a user wouldn't know what `Invalid token` means. If you could create a more helpful message that would be nice. Thanks.